dynamyc_list/list.cpp

```diff
@@ -0,0 +1,417 @@
+#include "list.h"
+
+/* Initializes list */
+void StringListInit(char*** list)
+{
+	if (*list)
+	{
+		StringListDestroy(list);
+	}
+	char** list_head = (char**)malloc(2 * sizeof(char*));
+
+	*list = list_head;//set pointer to head
+
+	list_head[0] = NULL;
+	list_head[1] = NULL;
+}
+
+/* Destroy list and set pointer to NULL. */
+void StringListDestroy(char*** list)
+{
+	char** current_elem = *list;
+	while (current_elem != NULL)
+	{
+		char** next_elem = (char**)current_elem[1];//get next element
+
+		free(current_elem[0]);//delete string in current_element
+		current_elem[0] = NULL;
+		free(current_elem);//delete current element
+		current_elem = NULL;
+
+		current_elem = next_elem;//go to next element
+	}
+	*list = NULL;
```

Comment:
You need call free(*list); before *list = NULL; it will be memory leak

Reply:
*list is a list head and it's memory was deleted on 1 iteration.

Reply:
*list remember list's head and i delete it on first iteration, but *list don't set to NULL
Because it i set *list to NULL in the function's end

> You need call free(*list); before *list = NULL; it will be memory leak